libraries/conf.rb

```diff
@@ -98,7 +98,7 @@ def whyrun_supported?
         execute "#{r.name}.#{r.conf_type}-systemd-reload" do
           command 'systemctl daemon-reload'
           action :nothing
-          only_if { r.is_a?(Chef::Resource::SystemdUnit) }
+          only_if { r.is_a?(Chef::Resource::SystemdUnit) && r.auto_reload }
           subscribes :run, "file[#{conf_path}]", :immediately
```

Comment:
> 12.5 event handler dsl

interesting, haven't seen this before... have to go do some reading.

> a delayed notification to a resource that does the daemon-reload might be easier

it's definitely the easiest way to do it, but would require users putting it in the run-list via an include_recipe, as opposed to us doing something that doesn't require coordination from the user. not a show-stopper, but not ideal either.